localization/pose_estimator/ndt_scan_matcher/ndt_omp/CMakeLists.txt

```diff
@@ -1,70 +1,81 @@
-if (NOT (${uppercase_CMAKE_BUILD_TYPE} STREQUAL "RELEASE"))
-  message(WARNING
-    "CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} may cause a serious performance degradation. Please configure with -DCMAKE_BUILD_TYPE=Release.")
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
 endif()
 
-find_package(catkin REQUIRED COMPONENTS
-  roscpp
-  pcl_ros
-)
+set(CMAKE_BUILD_TYPE "Release")
 
+find_package(ament_cmake REQUIRED)
+find_package(PCL REQUIRED COMPONENTS common filters kdtree registration io)
 find_package(OpenMP)
+
 if (OPENMP_FOUND)
   set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS}")
   set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
 endif()
 
-catkin_package(
-  INCLUDE_DIRS include
-  LIBRARIES ndt_omp
-  CATKIN_DEPENDS
-)
-
 ###########
 ## Build ##
 ###########
 
-include_directories(
-  include
-  ${catkin_INCLUDE_DIRS}
-)
-
 add_library(ndt_omp
   src/ndt_omp/voxel_grid_covariance_omp.cpp
   src/ndt_omp/ndt_omp.cpp
 )
 
-add_executable(align
-  apps/align.cpp
+target_include_directories(ndt_omp
+  PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include> 
 )
-add_dependencies(align
-  ndt_omp
+
+target_compile_options(ndt_omp
+  PRIVATE
+    # -mavx causes a lot of errors!!
+    -msse -msse2 -msse3 -msse4 -msse4.1 -msse4.2
 )
-# pcl 1.7 causes a segfault without -O2 (or -O3) flag
-target_compile_options(ndt_omp PUBLIC
-  $<$<CONFIG:Debug>:-O2 -g>
+
+set(NDT_OMP_DEPENDENCIES
+  PCL
+  OpenMP
 )
-target_link_libraries(align
-  ${catkin_LIBRARIES}
-  ndt_omp
+
+ament_target_dependencies(ndt_omp ${NDT_OMP_DEPENDENCIES}) 
+ament_export_targets(export_ndt_omp HAS_LIBRARY_TARGET)
+
+add_executable(align
+  apps/align.cpp
 )
 
-install(TARGETS ndt_omp
-  ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
+# Strangely, for an executable, ament_target_dependencies(align ${NDT_OMP_DEPENDENCIES})
+# gives a linker error from PCL
+ament_target_dependencies(align PUBLIC OpenMP)
+# So we include PCL manually
+target_include_directories(align PUBLIC ${PCL_INCLUDE_DIRS})
+target_link_libraries(align PUBLIC ${PCL_LIBRARIES})
+target_compile_definitions(align PUBLIC ${PCL_DEFINITIONS})
+target_link_directories(align PUBLIC ${PCL_LIBRARY_DIRS})
+# Also link ndt_omp separately because ament_target_dependencies doesn't support
+# depending on a library target from the same package
+target_link_libraries(align PUBLIC ndt_omp)
+
+ament_export_dependencies(${NDT_OMP_DEPENDENCIES})
+
+install(
+  DIRECTORY include/
+  DESTINATION include
 )
 
-## Install project namespaced headers
-install(DIRECTORY include/${PROJECT_NAME}/
-  DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
+install(
+  TARGETS ndt_omp
+  EXPORT export_ndt_omp
+  LIBRARY DESTINATION lib
+  ARCHIVE DESTINATION lib
+  RUNTIME DESTINATION bin
+  INCLUDES DESTINATION include
 )
+
+ament_package()
```

Comment:
can't you do `ament_package_auto()` instead of the long `install(...)` commands?

Reply:
I'm wary of mixing the two. I've tried it before and ran into problems.